### PR TITLE
fix(ci): Fix publishing of public docker images during releases of server pipelines

### PR DESCRIPTION
**Pull request by alexvy86**
## Description

The stage that publishes the public versions of docker images for the server components is currently broken ([example](https://dev.azure.com/fluidframework/internal/_build/results?buildId=273242&view=logs&s=2ff3421f-f27a-5211-cd8c-afeb9905f8f6), msft internal). The problem is that `$(containerTag)` is not getting replaced with a real value. This PR adjusts the server pipelines to address the issue.

I'm pretty confident the problem started [when we started using the 1ES pipeline templates](https://github.com/microsoft/FluidFramework/pull/19382). Before that, [the build-docker-service.yml template had variables at the root of the file](https://github.com/seanimam/FluidFramework/blob/f14abf74bd4474275b0ba0553728e543c13e06f3/tools/pipelines/templates/build-docker-service.yml#L100), so all stages in that file had access to them. When we moved to 1ES pipeline templates, those variables moved to be specified in the build stage and the publish stages lost access to them, which results in the `$(containerTag)` variable that the publish stage is trying to use to not getting replaced by ADO.

Changes in this PR:
- Remove the `include-publish-docker-service.yml` template (and CI/PR triggers off it) which is pretty redundant, only importing another file with no special logic.
- Introduces a new template `include-vars-docker.yml` that defines the variables which are relevant to pipelines that produce docker images. Almost a line-by-line copy of the variables that used to be in the build stage of `build-docker-service.yml`, with a few adjustments:
  - `tagNameOnReleaseImage` was removed from the list of variables and its logic is now inlined (and simplified) in the `publish_docker_official` stage of `build-docker-service.yml`.
  - Removed the bit that set `latestContainerTag` in the case of the `next` branch, since we don't have a `next` branch anymore.
  - The new template is now used in both the build stage and the `publish_docker_official` (the one that fails today). No other stages seem to need the new variables.
- Fix a small typo `1 -> i`

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

We cannot really test this effectively because it requires a build that publishes docker images to the public registry, which we only do during actual releases. I ran [a build of a test pipeline](https://dev.azure.com/fluidframework/internal/_build/results?buildId=294953&view=results) which succeeded, but as mentioned it doesn't cover all the changes from this PR.

### CODE REVIEW

File: tools/pipelines/templates/build-docker-service.yml [hunk 1]
```diff
@@ -575,10 +549,26 @@ extends:
...
+            environment: container-registry-public
+            # $(containerRegistryConnection) and $(containerRegistryUrl) come from the container-registry-info variable
+            # group and need to be specified as runtime variables (variables from variable groups apparently are never
+            # available "statically" at parse/compile time, so can't be used with template-expression syntax ( '${{ }}' )).
+            containerRegistry: $(containerRegistryConnection)
+            containerRegistryUrl: $(containerRegistryUrl)
+            containerTag: $(containerTag)
+            containerName: public/${{ parameters.containerName }}
+            ${{ if eq(parameters.releaseKind, 'docker') }}:
+              tagName: ${{ parameters.tagName }}
```
Comment (alexvy86) [inline tools/pipelines/templates/build-docker-service.yml:574]: This is basically a slightly adjusted copy-paste of what used to be in `include-publish-docker-service.yml`.

File: tools/pipelines/templates/build-docker-service.yml [hunk 2]
```diff
@@ -575,10 +549,26 @@ extends:
...
+      - stage: publish_docker_official
+        dependsOn: build
+        displayName: Publish Official Docker Image
+        condition: and(succeeded(), eq(variables['release'], 'release'))
+        variables:
+        - template: /tools/pipelines/templates/include-vars-docker.yml@self
+          parameters:
+            containerName: ${{ parameters.containerName }}
+            shouldPushDockerImage: ${{ parameters.shouldPushDockerImage }}
+            testBuild: ${{ parameters.testBuild }}
```
Comment (seanimam) [inline tools/pipelines/templates/build-docker-service.yml:561]: Do you need to include this template again here if it's already defined above / included in this file itself??

Comment (alexvy86) [reply tools/pipelines/templates/build-docker-service.yml:561]: Yes, when including another template you need to pass any parameters explicitly. The fact that this file takes a parameter with the same name doesn't matter when including another one.

**Review by seanimam — APPROVED** [review]

**Review by jason-ha — APPROVED** [review]
I am approving for infra. Relying on Sean's review for reasonable sanity checking.

**Pull request merged**